src/Chainweb/Miner/Coordinator.hs

```diff
@@ -134,43 +162,32 @@ newtype PrevTime = PrevTime BlockCreationTime
 
 data ChainChoice = Anything | TriedLast ChainId | Suggestion ChainId
 
--- | A `Miner`'s status. Will be `Primed` if defined in the nodes `miners` list.
--- This affects whether to serve the Miner primed payloads.
---
-data MinerStatus = Primed Miner | Plebian Miner
-
-minerStatus :: MinerStatus -> Miner
-minerStatus (Primed m) = m
-minerStatus (Plebian m) = m
-
 -- | Construct a new `BlockHeader` to mine on.
 --
 newWork
     :: LogFunction
     -> ChainChoice
-    -> MinerStatus
+    -> Miner
     -> WebBlockHeaderDb
         -- ^ this is used to lookup parent headers that are not in the cut
         -- itself.
     -> PactExecutionService
     -> TVar PrimedWork
     -> Cut
     -> IO (T2 WorkHeader PayloadData)
-newWork logFun choice eminer hdb pact tpw c = do
+newWork logFun choice eminer@(Miner mid _) hdb pact tpw c = do
 
     -- Randomly pick a chain to mine on, unless the caller specified a specific
     -- one.
     --
     cid <- chainChoice c choice
     logFun @T.Text Debug $ "newWork: picked chain " <> sshow cid
 
-    mr <- case eminer of
-        Primed m -> primed m cid <$> readTVarIO tpw
-        Plebian m -> do
-            logFun @T.Text Debug
-                $ "newWork: call getCutExtension for chain "
-                <> sshow cid <> " on cut " <> cutIdToTextShort (_cutId c)
-            public m cid
+    PrimedWork pw <- readTVarIO tpw
```

Comment:
We should probably double check, that the payload is valid for the cut, i.e. that it is mined on top of the most recent block for the respective chain. For that the parent hash (or just the block height might be enough?) should be included in the PrimedWork value.